src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -106,188 +107,196 @@ void EnCow_Init(Actor* thisx, PlayState* play) {
-            Collider_InitCylinder(play, &this->colliders[1]);
-            Collider_SetCylinder(play, &this->colliders[1], &this->actor, &sCylinderInit);
-            func_809DEE9C(this);
-            this->actionFunc = func_809DF96C;
+            Collider_InitCylinder(play, &this->colliders[EN_COW_COLLIDER_FRONT]);
+            Collider_SetCylinder(play, &this->colliders[EN_COW_COLLIDER_FRONT], &this->actor, &sCylinderInit);
+            Collider_InitCylinder(play, &this->colliders[EN_COW_COLLIDER_REAR]);
+            Collider_SetCylinder(play, &this->colliders[EN_COW_COLLIDER_REAR], &this->actor, &sCylinderInit);
+            EnCow_SetColliderPos(this);
+            this->actionFunc = EnCow_Idle;
             if (play->sceneId == SCENE_LINKS_HOUSE) {
                 if (!LINK_IS_ADULT) {
                     Actor_Kill(&this->actor);
                     return;
                 }
-                if (!GET_EVENTCHKINF(EVENTCHKINF_1E)) {
+                if (!GET_EVENTCHKINF(EVENTCHKINF_HORSE_RACE_COW_UNLOCK)) {
                     Actor_Kill(&this->actor);
                     return;
                 }
             }
-            Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_EN_COW, this->actor.world.pos.x,
-                               this->actor.world.pos.y, this->actor.world.pos.z, 0, this->actor.shape.rot.y, 0, 1);
-            this->unk_278 = Rand_ZeroFloat(1000.0f) + 40.0f;
-            this->unk_27A = 0;
+            Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_EN_COW,
+                               this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                               0, this->actor.shape.rot.y, 0, EN_COW_TYPE_TAIL);
+            this->animationTimer = Rand_ZeroFloat(1000.0f) + 40.0f;
+            this->breathTimer = 0;
             this->actor.targetMode = 6;
-            DREG(53) = 0;
+            R_PLAYED_EPONAS_SONG = false;
             break;
-        case 1:
-            SkelAnime_InitFlex(play, &this->skelAnime, &gCowTailSkel, NULL, this->jointTable, this->morphTable, 6);
+        case EN_COW_TYPE_TAIL:
+            SkelAnime_InitFlex(play, &this->skelAnime, &gCowTailSkel, NULL,
+                               this->jointTable, this->morphTable, COW_TAIL_LIMB_MAX);
             Animation_PlayLoop(&this->skelAnime, &gCowTailIdleAnim);
-            this->actor.update = func_809DFE98;
-            this->actor.draw = func_809E0070;
-            this->actionFunc = func_809DFA84;
-            func_809DEF94(this);
+            this->actor.update = EnCow_UpdateTail;
+            this->actor.draw = EnCow_DrawTail;
+            this->actionFunc = EnCow_IdleTail;
+            EnCow_SetTailPos(this);
             this->actor.flags &= ~ACTOR_FLAG_0;
-            this->unk_278 = ((u32)(Rand_ZeroFloat(1000.0f)) & 0xFFFF) + 40.0f;
+            this->animationTimer = ((u32)(Rand_ZeroFloat(1000.0f)) & 0xFFFF) + 40.0f;
```

Comment:
```suggestion
            this->animationTimer = (u16)Rand_ZeroFloat(1000.0f) + 40.0f
```
Also works and seems cleaner.